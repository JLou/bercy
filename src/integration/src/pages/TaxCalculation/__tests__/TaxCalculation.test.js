import React from 'react';
import { render } from '@testing-library/react';
import { TaxCalculation } from '../TaxCalculation';

describe('TaxCalculation', () => {
  it('Renders TaxCalculation component without crashing', () => {
    const { asFragment } = render(<TaxCalculation />);
    expect(asFragment()).toMatchSnapshot();
  });
});
