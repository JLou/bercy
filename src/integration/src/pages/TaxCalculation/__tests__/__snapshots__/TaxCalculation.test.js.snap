// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaxCalculation Renders TaxCalculation component without crashing 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="tax-calculation"
    >
      <form
        class="af-form tax-form"
      >
        <h1
          class="af-title--content"
        >
          Formulaire
        </h1>
        <div
          class="flex-container"
        >
          <dl>
            <dt>
              Nombre adulte :
            </dt>
            <dd>
              <input
                class="af-form__input-text"
                id="adultnumber"
                label="test"
                name="adultnumber"
                type="text"
                value=""
              />
            </dd>
          </dl>
          <dl>
            <dt>
              Montant salaire :
            </dt>
            <dd>
              <input
                class="af-form__input-text"
                id="salaryamount"
                label="test"
                name="salaryamount"
                type="text"
                value=""
              />
            </dd>
          </dl>
          <dl>
            <dt>
              Nombre enfant :
            </dt>
            <dd>
              <input
                class="af-form__input-text"
                id="numberofchildren"
                label="test"
                name="numberofchildren"
                type="text"
                value=""
              />
            </dd>
          </dl>
          <dl>
            <dt>
              Année :
            </dt>
            <dd>
              <div
                class="af-form__select-container"
              >
                <select
                  class="af-form__input-select"
                  name="year"
                >
                  <option
                    value="choice1"
                  >
                    2019
                  </option>
                  <option
                    value="choice2"
                  >
                    2020
                  </option>
                </select>
                <span
                  class="glyphicon glyphicon-menu-down"
                />
              </div>
            </dd>
          </dl>
        </div>
        <div
          class="af-form__group af-form__btn"
        >
          <button
            class="btn af-btn af-btn--hasiconLeft"
            id="validation-button"
          >
            <span
              class="af-btn__text"
            >
              Calculer
            </span>
            <i
              class="glyphicon glyphicon-stats"
            />
          </button>
        </div>
        <h1
          class="af-title--content"
        >
          Résultat
        </h1>
        <div
          class="tax-result"
        >
          <dl
            class="af-restitution__listdef"
          >
            <dt
              class="af-restitution__listdef-item"
            >
              <span
                class="af-restitution__text"
              >
                Taux d'imposition
              </span>
            </dt>
            <dd
              class="af-restitution__listdef-value"
            >
              ***
            </dd>
          </dl>
          <dl
            class="af-restitution__listdef"
          >
            <dt
              class="af-restitution__listdef-item"
            >
              <span
                class="af-restitution__text"
              >
                Montant impôt
              </span>
            </dt>
            <dd
              class="af-restitution__listdef-value"
            >
              ****
            </dd>
          </dl>
          <dl
            class="af-restitution__listdef"
          >
            <dt
              class="af-restitution__listdef-item"
            >
              <span
                class="af-restitution__text"
              >
                Nombre de part
              </span>
            </dt>
            <dd
              class="af-restitution__listdef-value"
            >
              ****
            </dd>
          </dl>
        </div>
      </form>
      <div
        class="maximum-ceiling"
      >
        <article
          class="af-panel"
        >
          <header
            class="af-panel__header"
          >
            <span
              class="af-panel__title"
            >
              Taux d'imposition par année
            </span>
            <div
              class="af-form__select-container"
            >
              <select
                class="af-form__input-select"
                name="name"
              >
                <option
                  value="choice1"
                >
                  2019
                </option>
                <option
                  value="choice2"
                >
                  2020
                </option>
              </select>
              <span
                class="glyphicon glyphicon-menu-down"
              />
            </div>
          </header>
          <section
            class="af-panel__content"
          >
            <table
              class="af-table"
            >
              <thead
                class="af-table-thead"
              >
                <tr
                  class="af-table__tr"
                >
                  <th
                    class="af-table__th"
                  >
                    <span
                      class="af-table-th-content"
                    >
                      Tranches
                    </span>
                  </th>
                  <th
                    class="af-table__th"
                  >
                    <span
                      class="af-table-th-content"
                    >
                      Taux d'imposition
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                class="af-table__body"
              >
                <tr
                  class="af-table__tr"
                >
                  <td
                    class="af-table__cell"
                  >
                    <span
                      class="af-table-body-content"
                    >
                      jusqu'à 9 964€
                    </span>
                  </td>
                  <td
                    class="af-table__cell"
                  >
                    <b>
                      0%
                    </b>
                  </td>
                </tr>
                <tr
                  class="af-table__tr"
                >
                  <td
                    class="af-table__cell"
                  >
                    <span
                      class="af-table-body-content"
                    >
                      de 9964 à 27519
                    </span>
                  </td>
                  <td
                    class="af-table__cell"
                  >
                    <b>
                      14%
                    </b>
                  </td>
                </tr>
              </tbody>
            </table>
          </section>
        </article>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
